Code/Framework/AzFramework/AzFramework/Device/DeviceAttributeRAM.cpp

```diff
@@ -28,9 +29,10 @@ namespace AzFramework
         return AZStd::any(m_valueInGiB);
     }
 
-    bool DeviceAttributeRAM::Evaluate([[maybe_unused]] AZStd::string_view rule) const
+    bool DeviceAttributeRAM::Evaluate(AZStd::string_view rule) const
     {
-        return false;
+        auto regex = AZStd::regex(rule.data(), rule.size());
+        return AZStd::regex_match(m_value, regex);

```

Comment:
The `Evaluate` methods between `DeviceAttributeRAM` and `DeviceAttributeDeviceModel` look nearly identical. Can the regex related functions be grouped together into a single definition?

Reply:
Yup, I plan on using a common eval function at some point when we have more than 2 attributes.